src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/CachedModelMetadata.cs

```diff
@@ -82,6 +83,17 @@ protected CachedModelMetadata(CachedModelMetadata<TPrototypeCache> prototype, Fu
             PrototypeCache = prototypeCache;
         }
 
+        // Sealing for consistency with other properties.
+        // ModelMetadata already caches the collection and we have no use case for a ComputeAdditionalValues() method.
+        /// <inheritdoc />
+        public sealed override IDictionary<string, object> AdditionalValues
```

Comment:
IMO this whole feature seems kinda superfluous without adding a way to do this via attributes